src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java

```diff
@@ -429,6 +429,11 @@ public void updateJobStatus(String jobId, JobStatus status, HashMap<String, Stri
                             .log("Job status updated rejected");
                     gotResponse.completeExceptionally(new Exception(response.message));
                 });
+
+        // Truncate status detail map values longer than the 1024 characters limit
+        statusDetailsMap.entrySet().forEach(e -> statusDetailsMap.put(e.getKey(),
+                e.getValue().substring(0, Math.min(e.getValue().length() - 1, 1023))));
```

Comment:
is the min really needed? If you just use 1023 does substring complain if the string is shorter?

Can you add a Util for truncating to a specific length and including ellipses so that we know it was truncated?

Reply:
also, this needs to be length, not length-1.